### PR TITLE
New Recipe: Track Customer Journey

### PR DESCRIPTION
**Pull request by ybyzek**
### New recipe checklist

- [x] Full code validated in Confluent Cloud
- [x] SQL style/syntax consistent to existing recipes
- [x] Validate presentation with `python3 -m mkdocs serve`
- [x] File structure and content follow [template](https://github.com/confluentinc/ksqldb-recipes/tree/main/template)
- [x] Added recipe to [navigation](https://github.com/confluentinc/ksqldb-recipes/blob/main/mkdocs.yml)
- [x] SQL style consistent to existing recipes
- [x] Spell check, e.g. using `aspell`

### CODE REVIEW

File: docs/customer-360/customer-journey/process.sql [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  KAFKA_TOPIC = 'pages',
+  PARTITIONS = 6
+);
+
+-- Create stateful table with list of pages visited by each customer
+-- Pages are added to an Array using the `COLLECT_LIST` function
+CREATE TABLE pages_per_customer WITH (KAFKA_TOPIC = 'pages_per_customer') AS
+SELECT
+  customer,
+  COLLECT_LIST(page) AS page_list,
```
Comment (ybyzek) [inline docs/customer-360/customer-journey/process.sql:20]: @MichaelDrogalis this recipe uses `COLLECT_LIST` as we discussed in a separate forum.  Do you think this set of ksqlDB statements is good enough, or any suggestions on how we can extend this recipe in more interesting ways

Comment (MichaelDrogalis) [reply docs/customer-360/customer-journey/process.sql:20]: Could also be interesting to include a `COUNT_DISTINCT` on the number of pages someone's visited, too. But LGTM!

Comment (ybyzek) [reply docs/customer-360/customer-journey/process.sql:20]: @MichaelDrogalis great idea, added in latest commit

**Review by fifthposition — APPROVED** [review]
LGTM (only reviewed Markdown content). Added a commit with minor edits.

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -9,5 +9,4 @@
 - [ ] Validate presentation with `python3 -m mkdocs serve`
 - [ ] File structure and content follow [template](https://github.com/confluentinc/ksqldb-recipes/tree/main/template)
 - [ ] Added recipe to [navigation](https://github.com/confluentinc/ksqldb-recipes/blob/main/mkdocs.yml)
-- [ ] SQL style consistent to existing recipes
```
Comment (mjsax) [inline .github/pull_request_template.md:12]: Why is this removed?

File: docs/customer-360/customer-journey/index.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
+---
+seo:
+  title: Track a Customer Journey
+  description: This recipe demonstrates how to use ksqlDB to track a customer journey through web pages online.
+---
+
+# Track a Customer Journey
+
+Companies with online websites, and customers that browse those websites, want to know which web pages their customers have visited.
```
Comment (mjsax) [inline docs/customer-360/customer-journey/index.md:9]: > online websites

Are not all websites online?

File: docs/customer-360/customer-journey/process.sql [hunk 4]
```diff
@@ -0,0 +1,24 @@
+SET 'auto.offset.reset' = 'earliest';
+
+-- Create stream of pages
+CREATE STREAM pages (
+  customer INTEGER,
+  time BIGINT,
+  page_id STRING,
```
Comment (mjsax) [inline docs/customer-360/customer-journey/process.sql:7]: Just wondering: do we prefer `STRING` or `VARCHAR` (and are all recipes consistent -- or maybe it does not matter)?

(Personally I am very confused that we even support `STRING` as synonym to `VARCHAR` which seems to be the "proper" SQL type name?)

File: docs/customer-360/customer-journey/process.sql [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+  page STRING
+) WITH (
+  VALUE_FORMAT = 'JSON',
+  KAFKA_TOPIC = 'pages',
+  PARTITIONS = 6
```
Comment (mjsax) [inline docs/customer-360/customer-journey/process.sql:12]: If we setup a connector, the topic should already exist and we don't need to specify the number of partitions?

(I leave the same comment on the other PR review -- not I am wondering about the case when users use the manual `INSERT INTO` path... maybe we can keep it but clarify that it only applies is the topic does not exist yet?)

File: docs/customer-360/customer-journey/process.sql [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+);
+
+-- Create stateful table with Array of pages visited by each customer, using the `COLLECT_LIST` function
+-- Get `COUNT_DISTINCT` page IDs
```
Comment (mjsax) [inline docs/customer-360/customer-journey/process.sql:16]: Not sure if important, but should we point out that `COUNT_DISTINCT` is not exact, but probabilistic?

File: .github/pull_request_template.md (hunk 2 above)
Comment (ybyzek) [reply .github/pull_request_template.md:12]: It's a duplicate of line 8

File: mkdocs.yml [hunk 7]
```diff
@@ -16,6 +16,7 @@ nav:
     - Understand user behavior with clickstream data: customer-360/clickstream/index.md
     - Notify passengers of flight updates: customer-360/aviation/index.md
     - Build customer loyalty programs: customer-360/loyalty-rewards/index.md
+    - Track customer journey: customer-360/customer-journey/index.md
```
Comment (ybyzek) [inline mkdocs.yml:19]:
```suggestion
    #- Track customer journey: customer-360/customer-journey/index.md
```

File: docs/customer-360/customer-journey/index.md (hunk 3 above)
Comment (sophiajiangg) [reply docs/customer-360/customer-journey/index.md:9]: @ybyzek @MichaelDrogalis this recipe seems very similar to the "Understand user behavior with clickstream data." Is the difference that this one is just looking at what pages were visited and how long someone stayed ont he page? VS. Clickstream is looking at what a customer journey might be through the web. 

Could you share with me which customer or how this use case was used. To second @mjsax's question, I'm not 100% clear why an org would just want to know what pages an individual customer visited. I could potentially see the use case for seeing which webpage is most popular.

Comment (rspurgeon) [reply docs/customer-360/customer-journey/index.md:9]: > @ybyzek @MichaelDrogalis this recipe seems very similar to the "Understand user behavior with clickstream data." Is the difference that this one is just looking at what pages were visited and how long someone stayed ont he page? VS. Clickstream is looking at what a customer journey might be through the web.
> 
> Could you share with me which customer or how this use case was used. To second @mjsax's question, I'm not 100% clear why an org would just want to know what pages an individual customer visited. I could potentially see the use case for seeing which webpage is most popular.

FWIW this has come up a few times now, a few comments from the associated issue: https://github.com/confluentinc/ksqldb-recipes/issues/106#issuecomment-1005933605

File: docs/customer-360/customer-journey/process.sql (hunk 6 above)
Comment (ybyzek) [reply docs/customer-360/customer-journey/process.sql:16]: @mjsax for users who want to read more about this function, https://github.com/confluentinc/ksqldb-recipes/pull/131/commits/c84823500d4b320f63994322df59ab656dd4947d adds a link to documentation.

File: docs/customer-360/customer-journey/process.sql (hunk 5 above)
Comment (ybyzek) [reply docs/customer-360/customer-journey/process.sql:12]: @mjsax will track separately in https://github.com/confluentinc/ksqldb-recipes/issues/156

File: docs/customer-360/customer-journey/process.sql (hunk 4 above)
Comment (ybyzek) [reply docs/customer-360/customer-journey/process.sql:7]: @mjsax will track separately in https://github.com/confluentinc/ksqldb-recipes/issues/146

File: docs/customer-360/customer-journey/index.md (hunk 3 above)
Comment (ybyzek) [reply docs/customer-360/customer-journey/index.md:9]: The queries are different and I think we should work to present them separately as well.  e.g. clickstream could focus on metadata like number of errors per minute and pages per minute, whereas customer journey could focus on individual web pages visited.
